addons/mail/static/src/messaging/entity/messaging_notification_handler/messaging_notification_handler.js

```diff
@@ -202,20 +213,23 @@ function MessagingNotificationHandlerFactory({ Entity }) {
          * @param {Object} data
          */
         _handleNotificationNeedaction(data) {
-            const message = this.env.entities.Message.insert(data);
-            const inboxMailbox = this.env.entities.Thread.mailboxFromId('inbox');
+            const message = this.env.entities.Message.insert(
+                this.env.entities.Message.convertData(data)
+            );
+            const inboxMailbox = this.env.entities.Thread.find(thread =>
+                thread.id === 'inbox' &&
+                thread.model === 'mail.box'
+            );
             inboxMailbox.update({ counter: inboxMailbox.counter + 1 });
             for (const thread of message.allThreads) {
                 if (
                     thread.channel_type === 'channel' &&
                     message.allThreads.includes(inboxMailbox)
                 ) {
-                    thread.update({
-                        message_needaction_counter: thread.message_needaction_counter + 1,
-                    });
+                    thread.update({ message_needaction_counter: thread.message_needaction_counter + 1 });
```

Comment:
I'm not sure if this is an issue or not in this case, but incrementing should be an atomic operation ideally (so like a command at update level). If we are assigning this value, but another part of the code is also assigning it just before/after, one or the other might get overwritten?
I'm not sure if async stuff can justify it or not, since JS is fundamentally not threaded it might never be an issue?